File: Week4/python-automation-test/pages/product_detail_page.py
from selenium.webdriver.common.by import By


class ProductDetailPage():
    def __init__(self, driver):
        self.driver = driver

        self.product_title_id = "productTitle"
        self.add_cart_button_id = "add-to-cart-button"

    def get_product_title(self):
        return self.driver.find_element(By.ID, self.product_title_id).text

    def click_add_cart(self):
        self.driver.find_element(By.ID, self.add_cart_button_id).click()

File: Week4/netcore-automation-test/pages/ProductDetailPage.cs
using System;
using OpenQA.Selenium;


namespace netcore_automation_test
{

    public class ProductDetailPage : HomePage
    {
        public ProductDetailPage(IWebDriver driver) : base(driver)
        {
            webDriver = driver;
        }

        public IWebElement ProductTitle => webDriver.FindElement(By.Id("productTitle"));
        public IWebElement AddCartButton => webDriver.FindElement(By.Id("add-to-cart-button"));

        public String GetProductTitle()
        {
            return GetTextElement(ProductTitle);
        }
        public CartPage ClickAddCart()
        {
            ClickElementWithJs(AddCartButton);
            return new CartPage(webDriver);
        }

    }

}